### ISSUE TITLE
UWP Support

### ISSUE THREAD

**Issue by archanox** [soywiz/cspspemu#91]
I know easier said than done, but with self publishing for UWP games on the XBOX ONE, this could be pretty cool.

I don't know if this is at all possible maybe due to the Dynarec, as I think there's a requirement to compile the UWP project to .Net native?

Food for thought!

**Comment by soywiz** [comment]
Hey Pierce! :D

Looks like a good idea. Though if it requires static compilation, indeed dynarec/JIT won't work. Maybe we can support an interpreted mode. Maybe a fast interpreter is fast enough to run at PSP speed; xbox one is pretty fast. Usually interpreters are one order of magnitude slower. But on my pc dynarec is already one magnitude faster :)

We will also need either managed directx for c# working on uwp, or a software renderer or both. Maybe we can reach full speed on a software renderer for 1x, but for higer resolutions we will definitely need directx too.

Do you know how to use directx on uwp and c#?
We also want to split some libraries in smaller parts, move to .net standard and publish them to nuget.

**Comment by UncleFirefox** [comment]
SharpDX my friend.... :)

**Comment by UncleFirefox** [comment]
The problem for XBOX ONE will be this, though: https://www.theverge.com/2017/4/6/15204138/microsoft-bans-game-emulators-from-the-windows-store

**Comment by soywiz** [comment]
Yeah. He says self publising. So no store. But one could install it on xbox one for indie developers probably or on computers.

**Comment by archanox** [comment]
Ahhh, I'm dumb. I forgot about the emulator ban.

Here's the self publishing I was referring to: https://www.theverge.com/2017/7/19/15997918/microsoft-xbox-live-creators-program-self-publish-games

**Comment by UncleFirefox** [comment]
Nonetheless I believe it's a good idea and since you can turn your XBOX ONE into dev mode easily we can leave some instructions on how to deploy to local machine for advanced users 👍 

https://www.windowscentral.com/how-activate-dev-mode-your-xbox-one-console

**Comment by UncleFirefox** [comment]
Interesting emulator recompilation for UWP: https://github.com/Aftnet/LibretroRT
